### ISSUE TITLE
lack of more .so files for other arch

### ISSUE THREAD

**Issue by LornTao** [Azure-Samples/Cognitive-Speech-STT-Android#54]
There are only two kinds of arch .so files, need more.

**Comment by wolfma61** [comment]
@LornTao - can you be more concrete about your requirements on Android? Platforms beside Android?

Some more background ...
We are aware that the client side support for Cognitive Service - Speech needs more attention. We are putting a dedicated team in place to address this. Next to ramping up support in the existing Cognitive Service Speech samples/libraries (incl. doc, high-pri fixes, etc), we are looking at enhancing the client side to have a much more cohesive API across platforms and cognitive services.  It will take a little more time to ramp up the team, but I'm confident we publish a road map in the near future (and I'm carefully omitting a concrete date here ;)

thx
Wolfgang

**Comment by rspq** [comment]
As others have reported, he means there's no ARM64 .so and your Android SDK is therefore unusable on the vast majority of current handsets. For those of us working in the Android world it's a fairly staggering omission.

**Comment by LornTao** [comment]
@rspq exactly right.

**Comment by wolfma61** [comment]
@LornTao, @rspq - we are looking into this now ... sorry for the inconvenience!

**Comment by raaaar** [comment]
Sorry, I'm confused, why is the SDK unusable? Have you tried the sample app and it didn't work? 
I'm totally able to run it on Nexus 5x with the following specs:

Processor       : AArch64 Processor rev 3 (aarch64)
[ro.product.cpu.abi]: [arm64-v8a]
[ro.product.cpu.abilist]: [arm64-v8a,armeabi-v7a,armeabi]

If the sample app does not work for you, what abi's does your device support?

**Comment by rspq** [comment]
I guess you are running as a 32-bit process, that's certainly a valid workaround. My understanding is that on 64-bit Linux (Android's base OS) a process can either be 64-bit or 32-bit, but you cannot load a 32-bit lib into a 64-bit process, or vice versa. I realise now that it is the presence of other 64-bit libs in my project that makes the absence of a 64-bit Bing STT lib a killer problem. If I remove all my 64-bit libs and go 32-bit only it might well work...

**Comment by raaaar** [comment]
Right, mixing 32- and 64-bits might be problematic. As @wolfma61  said, we're looking into this, but personally I wouldn't hold my breath for it (this, being a legacy client, might not have a highest prio). Meanwhile, take a look at the [REST sample](https://github.com/Azure-Samples/SpeechToText-REST) and see if it'll work for your app.

**Comment by wolfma61** [comment]
As I said before, we having a dedicated team on this right now, getting feature parity across all platforms and fixing annoying issues a work item ... we (including @raaar are getting up to speed and see what we can solve quickly and were we have to invest mid/long term)

**Comment by brandall76** [comment]
Just had my app crash due to this problem, so good to see it's actively being dealt with!

In my case, my app has separate architecture files for all libraries I use, therefore removing them to get this one to work isn't an option.

Here is how my build.gradle is set up:

```
            ndk {
                abiFilters 'armeabi-v7a', 'armeabi', 'arm64-v8a', 'x86', 'x86_64', 'mips', 'mips64'
            }
```
Just in case, here is a list of the [supported ABIs for Android](https://developer.android.com/ndk/guides/abis.html#sa).

**Comment by Toolenaar** [comment]
I am running into the same problem. The demo app works great. But I am trying to get it to work in a production app with lots of libraries that use 64bit. Is there an ETA on when this issue will be fixed ?

**Comment by raaaar** [comment]
Unfortunately, there's no ETA at this point. 

REST is meant as a quick and easy mechanisms to enable one-off experiments. To use a fully-functional speech reco, take a look at the web-socket based protocol. Here's a good [example](https://github.com/CatalystCode/SpeechToText-WebSockets-Java/).

**Comment by drummer3333** [comment]
The CatalystCode Websocket Repository is Closed down because "there is an official SDK for Java/Android".
Only that I need a Plain Java Version an I can not use this SDK.
